A15/PriorityQueue.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace A15;
+
+#region class PriorityQueue -----------------------------------------------------------------------
+/// <summary>
+/// A generic class depicts the typical functionality of the conventional class "Queue".
+/// Dequeue functionality returns the least object present in the queue. 
+/// </summary>
+public class PriorityQueue<T> where T : IComparable<T> {
+   #region Constructor ----------------------------------------------
+   public PriorityQueue () {
+      (mCount, mCapacity) = (0, 4);
+      mElements = new T[mCapacity];
+   }
+   #endregion
+
+   #region Methods --------------------------------------------------
+   /// <summary>Adds the given element into the queue</summary>
+   public void Enqueue (T t) {
+      if (mCount == mCapacity) {
+         mCapacity *= 2;
+         var tmp = new T[mCapacity];
+         for (int i = 0; i < mCount; i++) tmp[i] = mElements[i];
+         mElements = tmp;
+      }
+      mElements[mCount++] = t;
+      var rChild = mCount - 1;
+      while (rChild > 0) {
+         var lChild = rChild / 2; // Left child
+         var (lValue, rValue) = (mElements[lChild], mElements[rChild]);
+         if (rValue.CompareTo (lValue) >= 0) break;
+         (mElements[lChild], mElements[rChild]) = (rValue, lValue);
+         rChild = lChild;
+      }
+   }
+
+   /// <summary>Returns the element which is least object among the all other elements in the queue</summary>
+   public T Dequeue () {
+      var (index, parent) = (--mCount, 0);

```

Comment:
```suggestion
      var (max, parent) = (--mCount, 0);

```
max or last makes sense.